nomad/acl_endpoint.go

```diff
@@ -1964,3 +1964,20 @@ func (a *ACL) GetAuthMethods(
 		}},
 	)
 }
+
+func (a *ACL) WhoAmI(args *structs.GenericRequest, reply *structs.ACLWhoAmIResponse) error {
+
+	identity, err := a.srv.Authenticate(a.ctx, args.AuthToken)
+	if err != nil {
+		return err
+	}
+	args.SetIdentity(identity)
+
+	if done, err := a.srv.forward("ACL.WhoAmI", args, args, reply); done {
+		return err
+	}
+	defer metrics.MeasureSince([]string{"nomad", "acl", "whoami"}, time.Now())
+
+	reply.Identity = args.GetIdentity()
```

Comment:
What are the security implications of roundtripping this? I know that right now it's only reachable by devs and uniquely motivated users (because they'd need access to the rpc port, manually perform the rpc, and have a valid tls cert).

The only secret it returns is `ACLToken.SecretID`, and AFAICT the only way to have that set is to have passed that in.

So I think we're safe?

That being said perhaps we should at implement a `GetRedactedIdentity()` (or redact secrets from `GetIdentity()` by default and add an `UnsafeGetIdentity()` or similar if we callers need to reach in and use the ACLToken.SecretID directly).

As we strengthen our notion of identity more and more I think this (or similar) is going to be a really useful endpoint to expose.

Reply:
Typically we've done redaction at the structs->API layer, especially with the serialization extensions that Chris introduces on Node (which we also use on CSIVolume these days and I have an open issue to do for Job too).